cli/src/semgrep/commands/ci.py

```diff
@@ -465,7 +476,7 @@ def ci(
         all_targets=output_extra.all_targets,
         ignore_log=ignore_log,
         profiler=profiler,
-        filtered_rules=filtered_rules,
+        filtered_rules=[*blocking_rules, *nonblocking_rules, *cai_rules],
```

Comment:
This may be a bug from before this PR, but this function handles which findings are actually displayed to the user, right? If so, I am surprised to see cai_rules here, since those findings should be hidden. If it is right that cai_rules are included here, I find it odd that prev_scan_rules are not, so just want to check on what this is used for.

Reply:
Good catch there! I traced it down fully and looks like we only directly use ```filtered_rules``` for the scan timing metrics. 

However, I found this one snippet where ```filtered_rules (self.filtered_rules)``` is used to get ```num_rules```, which is part of the conditionals used to generate some user facing output lines (```suggestion_line``` and ```stats_line```). 

Really makes me think that the way to go is to exclude ```cai_rules``` as well and just have blocking and non-blocking rules. But again, I do think we also include (or at least don't intentionally exclude them) ```cai_rules``` in the scan summary part so maybe we should leave it as it is?

I am not sure. 😅 

![Screen Shot 2023-05-12 at 6 26 25 PM](https://github.com/returntocorp/semgrep/assets/48132553/0b69a928-5edd-4f36-ab38-06f285a34027)

Reply:
I'm pretty sure cai_rules are totally deprecated anyway, so should be safe to remove!